### PR TITLE
Area-Line rendering improvement

### PR DESCRIPTION
**Pull request by simzer**

### CODE REVIEW

File: src/chart/animator/planner.cpp [hunk 1]
```diff
@@ -251,6 +253,13 @@ void Planner::calcNeeded()
 
 	animNeeded[SectionId::y] = needVertical();
 	animNeeded[SectionId::x] = needHorizontal();
+
+	animNeeded[SectionId::connection] = anyMarker(
+		[&](const auto &source, const auto &target) -> bool {
+			return (bool)(source.prevMainMarkerIdx != target.prevMainMarkerIdx)
```
Comment (schaumb) [inline src/chart/animator/planner.cpp:259]: static_cast

File: src/chart/rendering/drawitem.cpp [hunk 2]
```diff
@@ -71,45 +70,73 @@ void drawItem::drawLines(const Styles::Guide &style,
...
+		auto containsSingle = 
+			options.shapeType.contains(Gen::ShapeType::rectangle) 
+			|| options.shapeType.contains(Gen::ShapeType::circle);
+
+		if (containsConnected)
+		{
+			if (containsSingle) 
+			{
+				auto lineIndex = 
+					options.shapeType.get(0).value == Gen::ShapeType::line
```
Comment (schaumb) [inline src/chart/rendering/drawitem.cpp:131]: Is this not the `Vizzu::Gen::isConnecting`?

File: src/chart/rendering/painter/drawline.cpp [hunk 3]
```diff
@@ -44,6 +46,23 @@ drawLine::drawLine(const Geom::Line &line,
 		canvas.setBrushColor(lineColor);
 		canvas.setLineColor(lineColor);
 
+		constexpr bool supportCurve = false;
+		if constexpr (supportCurve)
```
Comment (schaumb) [inline src/chart/rendering/painter/drawline.cpp:50]: What is the purpose of this?

File: src/chart/rendering/painter/painter.cpp [hunk 4]
```diff
@@ -13,11 +13,13 @@ void Painter::drawLine(const Geom::Line &line)
 
 void Painter::drawStraightLine(const Geom::Line &line,
     std::array<double, 2> widths,
+    double straightFactor,
     const Gfx::Color &endColor,
     const Gfx::Color &lineColor)
 {
 	Draw::drawLine(line,
 	    widths,
+		straightFactor,
```
Comment (schaumb) [inline src/chart/rendering/painter/painter.cpp:22]: space/tab

File: src/chart/rendering/items/connectingitem.cpp [hunk 5]
```diff
@@ -5,21 +5,27 @@
 using namespace Vizzu;
 using namespace Vizzu::Draw;
 
-ConnectingDrawItem::ConnectingDrawItem(const Gen::Marker &marker,
+ConnectingItem::ConnectingItem(const Gen::Marker &marker,
     const CoordinateSystem &coordSys,
     const Gen::Options &options,
+	const Styles::Chart &style,
```
Comment (schaumb) [inline src/chart/rendering/items/connectingitem.cpp:11]: space/tab

File: src/chart/rendering/painter/drawline.cpp (hunk 3 above)
Comment (simzer) [reply src/chart/rendering/painter/drawline.cpp:50]: This is how it should be implemented, but it makes a bug a lot worst, so we should not use it till the bug is not fixed.

File: src/chart/rendering/drawitem.cpp (hunk 2 above)
Comment (simzer) [reply src/chart/rendering/drawitem.cpp]: Yes, it is.

File: src/chart/rendering/painter/drawline.cpp (hunk 3 above)
Comment (schaumb) [reply src/chart/rendering/painter/drawline.cpp:50]: could you add a comment why it is in the code?

**Review by schaumb — APPROVED** [review]

File: src/chart/rendering/painter/drawline.cpp (hunk 3 above)
Comment (simzer) [reply src/chart/rendering/painter/drawline.cpp:50]: I have made an issue about this: https://trello.com/c/VtASDMJW/415-bug-vertical-polar-line-chart-should-have-curved-lines

Comment (simzer) [reply src/chart/rendering/painter/drawline.cpp:50]: I wouldn't put a comment into the code, but I could remove the switched off code part.

**Pull request merged**